`odo init --name <name> --devfile-path <path>` panics when a non-existent path is used

/kind bug

## What versions of software are you using?

**Operating System:**
Fedora 34

**Output of `odo version`:**
odo v2.5.0 (daa9c61a4)

## How did you run odo exactly?
`odo init --name aname --devfile-path /tmp/devfile.go`

## Actual behavior
```sh
➜  ls /tmp/devfile.yaml           
/tmp/devfile.yaml

➜ odo init --name aname --devfile-path /tmp/devfile.go
 ✗  Copying devfile from "/tmp/devfile.go" [65681ns]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xc0 pc=0x29030d3]

goroutine 1 [running]:
github.com/redhat-developer/odo/pkg/odo/cli/init.(*InitOptions).Run(0xc0003acc00, 0x344f820, 0xc0005a8040)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/cli/init/init.go:138 +0x273
github.com/redhat-developer/odo/pkg/odo/genericclioptions.GenericRun(0x34a0970, 0xc0003acc00, 0xc000838500, 0xc00089c0c0, 0x0, 0x4)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/genericclioptions/runnable.go:107 +0x4bf
github.com/redhat-developer/odo/pkg/odo/cli/init.NewCmdInit.func1(0xc000838500, 0xc00089c0c0, 0x0, 0x4)
        /home/pvala/go/src/github.com/redhat-developer/odo/pkg/odo/cli/init/init.go:188 +0x5e
github.com/spf13/cobra.(*Command).execute(0xc000838500, 0xc00089c080, 0x4, 0x4, 0xc000838500, 0xc00089c080)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:856 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0xc00013ec80, 0xc000122000, 0x2da0020, 0x45ea750)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:960 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:897
github.com/spf13/cobra.(*Command).ExecuteContext(...)
        /home/pvala/go/src/github.com/redhat-developer/odo/vendor/github.com/spf13/cobra/command.go:890
main.main()
        /home/pvala/go/src/github.com/redhat-developer/odo/cmd/odo/odo.go:70 +0x426
```

## Expected behavior
It should not panic. It should point out that the file does not exist.

## Any logs, error output, etc?

--- a/pkg/odo/cli/init/init.go
+++ b/pkg/odo/cli/init/init.go
@@ -134,7 +134,9 @@ func (o *InitOptions) Run() (err error) {
 	}
 
 	devfileObj, devfilePath, err := o.clientset.InitClient.SelectAndPersonalizeDevfile(o.flags, o.contextDir)
-
+	if err != nil {
+		return err
+	}
 	scontext.SetComponentType(o.ctx, component.GetComponentTypeFromDevfileMetadata(devfileObj.Data.GetMetadata()))
 
 	starterInfo, err := o.clientset.InitClient.SelectStarterProject(devfileObj, o.flags, o.clientset.FS, o.contextDir)

--- a/tests/integration/devfile/cmd_devfile_init_test.go
+++ b/tests/integration/devfile/cmd_devfile_init_test.go
@@ -2,6 +2,7 @@ package devfile
 
 import (
 	"io/ioutil"
+	"os"
 	"path/filepath"
 
 	. "github.com/onsi/ginkgo"
@@ -35,13 +36,21 @@ var _ = Describe("odo devfile init command tests", func() {
 		})
 		By("running odo init in a directory containing a devfile.yaml", func() {
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-registry.yaml"), filepath.Join(commonVar.Context, "devfile.yaml"))
+			defer os.Remove(filepath.Join(commonVar.Context, "devfile.yaml"))
 			err := helper.Cmd("odo", "init").ShouldFail().Err()
 			Expect(err).To(ContainSubstring("a devfile already exists in the current directory"))
 		})
+
 		By("running odo init in a directory containing a .devfile.yaml", func() {
 			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-registry.yaml"), filepath.Join(commonVar.Context, ".devfile.yaml"))
+			defer os.Remove(filepath.Join(commonVar.Context, ".devfile.yaml"))
 			err := helper.Cmd("odo", "init").ShouldFail().Err()
 			Expect(err).To(ContainSubstring("a devfile already exists in the current directory"))
+		})
+
+		By("running odo init with wrong path given to --devfile-path", func() {
+			err := helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", "/some/path/devfile.yaml").ShouldFail().Err()
+			Expect(err).To(ContainSubstring("no such file or directory"))
 		})
 	})
 

--- a/tests/integration/devfile/cmd_devfile_init_test.go
+++ b/tests/integration/devfile/cmd_devfile_init_test.go
@@ -48,9 +48,13 @@ var _ = Describe("odo devfile init command tests", func() {
 			Expect(err).To(ContainSubstring("a devfile already exists in the current directory"))
 		})
 
-		By("running odo init with wrong path given to --devfile-path", func() {
+		By("running odo init with wrong local file path given to --devfile-path", func() {
 			err := helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", "/some/path/devfile.yaml").ShouldFail().Err()
 			Expect(err).To(ContainSubstring("no such file or directory"))
+		})
+		By("running odo init with wrong URL path given to --devfile-path", func() {
+			err := helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", "https://github.com/path/to/devfile.yaml").ShouldFail().Err()
+			Expect(err).To(ContainSubstring("unable to download devfile: failed to retrieve"))
 		})
 	})
 
@@ -65,7 +69,7 @@ var _ = Describe("odo devfile init command tests", func() {
 				Expect(files).To(Equal([]string{"devfile.yaml"}))
 			})
 		})
-		When("using --devfile-path flag", func() {
+		When("using --devfile-path flag with a local devfile", func() {
 			var newContext string
 			BeforeEach(func() {
 				newContext = helper.CreateNewContext()
@@ -80,8 +84,17 @@ var _ = Describe("odo devfile init command tests", func() {
 				files := helper.ListFilesInDir(commonVar.Context)
 				Expect(files).To(Equal([]string{"devfile.yaml"}))
 			})
-
 		})
+		When("using --devfile-path flag with a URL", func() {
+			BeforeEach(func() {
+				helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", "https://raw.githubusercontent.com/odo-devfiles/registry/master/devfiles/nodejs/devfile.yaml").ShouldPass()
+			})
+			It("should copy the devfile.yaml file", func() {
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Equal([]string{"devfile.yaml"}))
+			})
+		})
+
 	})
 
 	When("running odo init from a directory with sources", func() {

--- a/tests/integration/devfile/cmd_devfile_init_test.go
+++ b/tests/integration/devfile/cmd_devfile_init_test.go
@@ -54,14 +54,33 @@ var _ = Describe("odo devfile init command tests", func() {
 		})
 	})
 
-	When("running odo init with valid flags", func() {
-		BeforeEach(func() {
-			helper.Cmd("odo", "init", "--name", "aname", "--devfile", "go").ShouldPass()
-		})
+	Context("running odo init with valid flags", func() {
+		When("using --devfile flag", func() {
+			BeforeEach(func() {
+				helper.Cmd("odo", "init", "--name", "aname", "--devfile", "go").ShouldPass().Out()
+			})
 
-		It("should download a devfile.yaml file", func() {
-			files := helper.ListFilesInDir(commonVar.Context)
-			Expect(files).To(Equal([]string{"devfile.yaml"}))
+			It("should download a devfile.yaml file", func() {
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Equal([]string{"devfile.yaml"}))
+			})
+		})
+		When("using --devfile-path flag", func() {
+			var newContext string
+			BeforeEach(func() {
+				newContext = helper.CreateNewContext()
+				newDevfilePath := filepath.Join(newContext, "devfile.yaml")
+				helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-registry.yaml"), newDevfilePath)
+				helper.Cmd("odo", "init", "--name", "aname", "--devfile-path", newDevfilePath).ShouldPass()
+			})
+			AfterEach(func() {
+				helper.DeleteDir(newContext)
+			})
+			It("should copy the devfile.yaml file", func() {
+				files := helper.ListFilesInDir(commonVar.Context)
+				Expect(files).To(Equal([]string{"devfile.yaml"}))
+			})
+
 		})
 	})
 

